### PR TITLE
Enable test hive-jdbc driver

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR enable test `hive-jdbc` driver.

### CODE REVIEW

File: sql/service/src/test/scala/org/apache/spark/sql/service/SparkThriftServer2Suites.scala [hunk 1]
```diff
@@ -866,10 +867,10 @@ abstract class SparkThriftJdbcTest extends SparkThriftServer2Test with JdbcTestH
        |hive.server2.thrift.http.path=cliservice;
        |spark.sql.thriftserver.transport.mode=http;
        |spark.sql.thriftserver.thrift.http.path=cliservice;
-       |${hiveConfList}#${hiveVarList}
+       |${hiveConfList}#${hiveVarList};${sparkConfList}
```
Comment (wangyum) [inline sql/service/src/test/scala/org/apache/spark/sql/service/SparkThriftServer2Suites.scala]: Connection URL Format：
```
jdbc:hive2://<host1>:<port1>,<host2>:<port2>/dbName;initFile=<file>;sess_var_list?hive_conf_list#hive_var_list
```

https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-ConnectionURLFormat

File: sql/service/src/test/scala/org/apache/spark/sql/service/SparkThriftServer2Suites.scala [hunk 2]
```diff
@@ -0,0 +1,1158 @@
...
+  SparkUtils.classForName(jdbcDriver)
+
+  private def jdbcUri = if (mode == ServerMode.http) {
+    s"""${jdbcUrlPrefix}localhost:$serverPort/
+       |default?
+       |hive.server2.transport.mode=http;
+       |hive.server2.thrift.http.path=cliservice;
+       |spark.sql.thriftserver.transport.mode=http;
+       |spark.sql.thriftserver.thrift.http.path=cliservice;
+       |${hiveConfList};${sparkConfList}#${hiveVarList}
```
Comment (wangyum) [inline sql/service/src/test/scala/org/apache/spark/sql/service/SparkThriftServer2Suites.scala]: Connection URL Format：
```
jdbc:hive2://<host1>:<port1>,<host2>:<port2>/dbName;initFile=<file>;sess_var_list?hive_conf_list#hive_var_list
```

https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-ConnectionURLFormat

File: sql/service/src/test/scala/org/apache/spark/sql/service/SparkThriftServer2Suites.scala [hunk 3]
```diff
@@ -0,0 +1,1156 @@
...
+        } finally {
+          statement.executeQuery("SET spark.sql.hive.thriftServer.async=true")
+        }
+      } finally {
+        ec.shutdownNow()
+      }
+    }
+  }
+
+  // TODO: Test custom FileFormat by `AddJarCommand`.
```
Comment (wangyum) [inline sql/service/src/test/scala/org/apache/spark/sql/service/SparkThriftServer2Suites.scala]: Will test custom FileFormat by `AddJarCommand` in future.

**Comment by juliuszsompolski** [comment]
Nice :+1: !

**Pull request merged**